plugin/src/main/java/com/epam/jenkins/deployment/sphere/plugin/utils/SubMetaDataExtractor.java

```diff
@@ -0,0 +1,59 @@
+package com.epam.jenkins.deployment.sphere.plugin.utils;
+
+import hudson.model.Result;
+import hudson.model.AbstractBuild;
+import hudson.scm.ChangeLogSet;
+import hudson.scm.ChangeLogSet.Entry;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.codehaus.jackson.map.ObjectMapper;
+
+import com.epam.jenkins.deployment.sphere.plugin.metadata.model.Commit;
+import com.epam.jenkins.deployment.sphere.plugin.metadata.model.SubMetaData;
+
+public class SubMetaDataExtractor {
+
+    public static String getMetaData(final AbstractBuild<?, ?> build) {
```

Comment:
method is with too complex logic, please refactor to several smaller methods with proper purpopse